core/src/main/java/io/opentelemetry/android/OpenTelemetryRumBuilder.java

```diff
@@ -279,6 +286,56 @@ public OpenTelemetryRum build() {
         InitializationEvents initializationEvents = InitializationEvents.get();
         applyConfiguration(initializationEvents);
 
+        BufferDelegatingLogExporter bufferDelegatingLogExporter = new BufferDelegatingLogExporter();
+
+        BufferDelegatingSpanExporter bufferDelegatingSpanExporter =
+                new BufferDelegatingSpanExporter();
+
+        SessionManager sessionManager =
+                SessionManager.create(timeoutHandler, config.getSessionTimeout().toNanos());
+
+        OpenTelemetrySdk sdk =
+                OpenTelemetrySdk.builder()
+                        .setTracerProvider(
+                                buildTracerProvider(
+                                        sessionManager, application, bufferDelegatingSpanExporter))
+                        .setLoggerProvider(
+                                buildLoggerProvider(
+                                        sessionManager, application, bufferDelegatingLogExporter))
+                        .setMeterProvider(buildMeterProvider(application))
+                        .setPropagators(buildFinalPropagators())
+                        .build();
+
+        otelSdkReadyListeners.forEach(listener -> listener.accept(sdk));
+
+        SdkPreconfiguredRumBuilder delegate =
+                new SdkPreconfiguredRumBuilder(
+                        application,
+                        sdk,
+                        timeoutHandler,
+                        sessionManager,
+                        config.shouldDiscoverInstrumentations(),
+                        getServiceManager());
+
+        // AsyncTask is deprecated but the thread pool is still used all over the Android SDK
+        // and it provides a way to get a background thread without having to create a new one.
+        android.os.AsyncTask.THREAD_POOL_EXECUTOR.execute(
```

Comment:
Selling the use of deprecated tools is tough, though I understand the concern about creating many threads in Android. If this PR gets merged I think I can help address this by creating a work for `PeriodicWorkService` that executes once. It's an internal tool that allows us to run stuff in the background by reusing a single thread. Alternatively, using the default Kotlin coroutine dispatcher as an executor via `ExecutorsKt.asExecutor(Dispatchers.getDefault()).execute(...);` could be another way to do the same but without having to use deprecated tools.

Reply:
Yes it's unfortunate. But like the comment mentions, the thread pool is actually still used in a lot of places, including by AndroidX and Android internally. It felt worthwhile to not spin up new threads.

I was under the impression that coroutines is not used in this library, otherwise I would have used the coroutine dispatchers.

Reply:
Got it, that's fair enough. I think it's fine to leave it as is for now.